addons/mail/static/src/models/message/message.js

```diff
@@ -197,6 +197,49 @@ function factory(dependencies) {
                 kwargs: kwargs,
             });
         }
+        /**
+         * Performs the `message_fetch` RPC on `mail.message`.
+         *
+         * @param {Array[]} domain
+         * @param {integer} [limit]
+         * @param {integer[]} [moderated_channel_ids]
+         * @param {Object} [context]
+         * @static
+         */
+        static async performRpcMessageFetch(domain, limit, moderated_channel_ids, context) {
+            const messagesData = await this.env.services.rpc({
+                model: 'mail.message',
+                method: 'message_fetch',
+                kwargs: {
+                    context,
+                    domain,
+                    limit,
+                    moderated_channel_ids,
+                },
+            }, { shadow: true });
+            const messages = this.env.models['mail.message'].insert(messagesData.map(
+                messageData => this.env.models['mail.message'].convertData(messageData)
+            ));
+            for (const message of messages) {
+                for (const thread of message.threads) {
+                    if (thread.model !== 'mail.channel' || thread.channel_type === 'channel') {
+                        // disabled on non-channel threads and
+                        // on `channel` channels for performance reasons
+                        continue;
+                    }
+                    thread.update({
+                        messageSeenIndicators: [[
+                            'insert',
+                            {
+                                id: this.env.models['mail.message_seen_indicator'].computeId(message.id, thread.id),
+                                message: [['link', message]],
+                            },
+                        ]],
+                    });
+                }
+            }
```

Comment:
Would put comment that these LOCs manually compute seen indicators.
Makes it easier to skip reading if we don't care, and also the comment above `continue;` is easier to get (I had to stare my screen for several seconds, wondering why skip processing of message fetch of "non-channel threads and 'channel' channels for performance reasons")

Reply:
ok